internal/internal_workflow_test.go

```diff
@@ -1387,6 +1390,240 @@ func (s *WorkflowUnitTest) Test_StaleGoroutinesAreShutDown() {
 	}
 }
 
+type testUpdateCallbacks struct {
+	AcceptImpl   func()
+	RejectImpl   func(err error)
+	CompleteImpl func(success interface{}, err error)
+}
+
+func (tuc *testUpdateCallbacks) Accept()          { tuc.AcceptImpl() }
+func (tuc *testUpdateCallbacks) Reject(err error) { tuc.RejectImpl(err) }
+func (tuc *testUpdateCallbacks) Complete(success interface{}, err error) {
+	tuc.CompleteImpl(success, err)
+}
+
+func TestDefaultUpdateHandler(t *testing.T) {
```

Comment:
I didn't use the suite here because workflows run from those utilities appear not to integrate properly with the coro dispatcher so whenever we hit a Go() or a GoNamed() the dispatcher would panic. Open to suggestions on how to integrabe better but tbh this test came out fairly clean.